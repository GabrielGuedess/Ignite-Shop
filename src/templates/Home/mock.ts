import { HomeProps } from '.';

export const mockHome: HomeProps = {
  products: [
    {
      id: '1',
      imageUrl:
        'https://d1wqzb5bdbcre6.cloudfront.net/d7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f/68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164',
      name: 'Camiseta X',
      price: 'R$ 79,00',
    },

    {
      id: '2',
      imageUrl:
        'https://d1wqzb5bdbcre6.cloudfront.net/d7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f/68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164',
      name: 'Camiseta Y',
      price: 'R$ 79,00',
    },

    {
      id: '3',
      imageUrl:
        'https://d1wqzb5bdbcre6.cloudfront.net/d7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f/68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164',
      name: 'Camiseta Z',
      price: 'R$ 79,00',
    },

    {
      id: '4',
      imageUrl:
        'https://d1wqzb5bdbcre6.cloudfront.net/d7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f/68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164',
      name: 'Camiseta A',
      price: 'R$ 79,00',
    },
  ],
};
