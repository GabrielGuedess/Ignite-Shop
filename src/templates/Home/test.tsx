import { render } from '@testing-library/react';

import { Home } from '.';
import { mockHome } from './mock';

describe('<Home />', () => {
  it('should render the home correctly', () => {
    const { container } = render(<Home products={mockHome.products} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
