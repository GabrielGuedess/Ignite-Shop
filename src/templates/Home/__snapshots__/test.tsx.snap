// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the home correctly 1`] = `
<div
  class="c-fIbhzJ keen-slider"
>
  <a
    class="c-jmYiOV keen-slider__slide"
    href="/product/1"
  >
    <img
      alt="camiseta"
      data-nimg="1"
      decoding="async"
      height="480"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75 2x"
      style="color: transparent;"
      width="520"
    />
    <footer
      class="c-fHCWwZ"
    >
      <strong
        class="c-eJYKcl"
      >
        Camiseta X
      </strong>
      <span
        class="c-cBOCLu"
      >
        R$ 79,00
      </span>
    </footer>
  </a>
  <a
    class="c-jmYiOV keen-slider__slide"
    href="/product/2"
  >
    <img
      alt="camiseta"
      data-nimg="1"
      decoding="async"
      height="480"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75 2x"
      style="color: transparent;"
      width="520"
    />
    <footer
      class="c-fHCWwZ"
    >
      <strong
        class="c-eJYKcl"
      >
        Camiseta Y
      </strong>
      <span
        class="c-cBOCLu"
      >
        R$ 79,00
      </span>
    </footer>
  </a>
  <a
    class="c-jmYiOV keen-slider__slide"
    href="/product/3"
  >
    <img
      alt="camiseta"
      data-nimg="1"
      decoding="async"
      height="480"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75 2x"
      style="color: transparent;"
      width="520"
    />
    <footer
      class="c-fHCWwZ"
    >
      <strong
        class="c-eJYKcl"
      >
        Camiseta Z
      </strong>
      <span
        class="c-cBOCLu"
      >
        R$ 79,00
      </span>
    </footer>
  </a>
  <a
    class="c-jmYiOV keen-slider__slide"
    href="/product/4"
  >
    <img
      alt="camiseta"
      data-nimg="1"
      decoding="async"
      height="480"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fd1wqzb5bdbcre6.cloudfront.net%2Fd7e0f05b16a7dd5c6bda9d2b8023377c8d169c9d44fb684b9afea972f3bfa90f%2F68747470733a2f2f66696c65732e7374726970652e636f6d2f6c696e6b732f4d44423859574e6a6446387854566857527a46435a557853526e6c33616a427466475a735833526c633352664f4770614f47747a5a4539616257313457484635527a427855566c75646b707230307049447645725164&w=1080&q=75 2x"
      style="color: transparent;"
      width="520"
    />
    <footer
      class="c-fHCWwZ"
    >
      <strong
        class="c-eJYKcl"
      >
        Camiseta A
      </strong>
      <span
        class="c-cBOCLu"
      >
        R$ 79,00
      </span>
    </footer>
  </a>
</div>
`;
